cmds/wifi/wifi.go

```diff
@@ -146,22 +156,69 @@ func generateConfig(a ...string) (conf []byte, err error) {
 	return
 }
 
-func main() {
-	// Service
-	var (
-		conf []byte
-		err  error
-	)
+// To prevent race condition, there should be only one
+// goroutine running connectWifiArbitrator at any one time
+func connectWifiArbitrator() {
+	for req := range ConnectReqChan {
+		if ConnectingEssid != "" {
```

Comment:
This is a race condition. You're going to need a better way to do this. You might want to run this with go test -race.
The problem is you are testing ConnectingEssid and setting it too.

Reply:
you are right, I'm wrong, this is fine.

Reply:
well, no, I still don't like it: how do you ensure that the goroutine writing the request is the one getting the answer? This is almost right but needs a bit of work.

Reply:
I pass in the channel that the goroutine writing the request will be listening on inside the message that is passed to ConnectReqChan (similar to what you said about the Plan9 system), this ensures that only the goroutine passing the check will receive a nill error and only it can see that. All other goroutine will receive a non-nil error

Reply:
sorry, I missed that!